File: frontend/src/types/note.ts
export type NoteStatus = "todo" | "ongoing" | "done";

export type Note = {
  id: string;
  title: string;
  content: string;
  status: NoteStatus;
  is_live: boolean;
  createdAt: string;
};

export type KanbanColumn = {
  id: NoteStatus;
  label: string;
};

export const KANBAN_COLUMNS: KanbanColumn[] = [
  { id: "todo", label: "Todo" },
  { id: "ongoing", label: "Ongoing" },
  { id: "done", label: "Done" },
];

File: backend/internal/models/note.go
package models

import "time"

type Note struct {
	ID        string    `json:"id"`
	UserID    string    `json:"user_id"`
	Title     string    `json:"title"`
	Content   string    `json:"content"`
	Status    string    `json:"status"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	IsLive    bool      `json:"is_live"`
}
